coffeecard/CoffeeCard.WebApi/Controllers/v2/ProductsController.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using CoffeeCard.Library.Services;
+using CoffeeCard.Library.Services.v2;
+using CoffeeCard.Library.Utils;
+using CoffeeCard.Models.DataTransferObjects.v2.Leaderboard;
+using CoffeeCard.Models.DataTransferObjects.v2.Product;
+using CoffeeCard.Models.Entities;
+using CoffeeCard.WebApi.Helpers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using IProductService = CoffeeCard.Library.Services.v2.IProductService;
+
+namespace CoffeeCard.WebApi.Controllers.v2
+{
+    /// <summary>
+    /// Controller for creating, changing, and deactivating a product
+    /// </summary>
+    [ApiController]
+    [Authorize]
+    [ApiVersion("2")]
+    [Route("api/v{version:apiVersion}/products")]
+
+    public class ProductsController : ControllerBase
+    {
+        private readonly IProductService _productService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProductsController"/> class.
+        /// </summary>
+        public ProductsController(IProductService productService)
+        {
+            _productService = productService;
+        }
+
+        /// <summary>
+        /// Adds a new product to the database.
+        /// </summary>
+        /// <param name="addProductRequest">The request containing the details of the product to be added and allowed user groups.</param>
+        /// <returns> The newly added product wrapped in a InitiateProductResponse object.</returns>
+        /// <response code="200">The request was successful, and the product was added.</response>
+        [HttpPost("add")]
+        [AuthorizeRoles(UserGroup.Board)]
+        [ProducesResponseType(typeof(ProductResponse), StatusCodes.Status200OK)]
+        public async Task<IActionResult> AddProduct(AddProductRequestWithUserGroups addProductRequest)
```

Comment:
Per RESTful design practices, I suggest the API should simply be

```
POST api/v2/products
```

That is remove the add.

Further reference https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design#organize-the-api-design-around-resources